### PR TITLE
Update RAJA and friends

### PR DESCRIPTION
**Pull request by white238**
sigh.

### CODE REVIEW

File: src/cmake/thirdparty/SetupAxomThirdParty.cmake [hunk 1]
```diff
@@ -32,7 +32,8 @@ if ((RAJA_DIR OR UMPIRE_DIR) AND NOT CAMP_DIR)
     message(FATAL_ERROR "CAMP_DIR is required if RAJA_DIR or UMPIRE_DIR is provided.")
 endif()
 
-# Note: Let Umpire find Camp, don't find it ourselves
+# Note: Let Umpire find Camp via camp_DIR, don't find it ourselves
+set(camp_DIR ${CAMP_DIR})
```
Comment (cyrush) [inline src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: Oh  man :-(

Comment (davidbeckingsale) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: This is the CMake convention! Don't shoot the messenger.

Comment (white238) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: I'm going to create a loop that will create every possible variation in capitalization for each TPL.... CaMP_DiR, cAmP_Dir...

Comment (davidbeckingsale) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: image

Comment (rhornung67) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: This frightens me, yet I cannot look away....

Comment (rhornung67) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: sounds like another item to add to @white238 's CMake/BLT best practices guide.

Comment (white238) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: Yes, we will provide guidance on the best practices and add a line to support all uppercase for backwards compatibility.

Comment (rhornung67) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: So all uppercase to support Fortran users? 😄

**Comment by white238** [comment]
Ready for review.

**Review by bmhan12 — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

File: src/cmake/thirdparty/SetupAxomThirdParty.cmake (hunk 1 above)
Comment (cyrush) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: well, the FORTRAN users, yes.

Comment (agcapps) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: > well, the FORTRAN users, yes.

*rim shot*

Comment (rhornung67) [reply src/cmake/thirdparty/SetupAxomThirdParty.cmake:36]: Should we require all Fortran users to start every line in their CMake files in column 7?  Sorry an old joke about Fortran 77.

**Pull request merged**